releasenotes/notes/control-flow-depth-size-b598a4eb9d8888eb.yaml

```diff
@@ -0,0 +1,15 @@
+---
+features:
+  - |
+    The circuit method :meth:`.QuantumCircuit.depth`, and the :class:`.DAGCircuit`
+    methods :meth:`~.DAGCircuit.depth` and :meth:`~.DAGCircuit.size` gained a
+    ``recurse`` keyword argument for use with control flow.  By default this is
+    ``False`` and will raise an error if control-flow is present, to avoid
+    nonsensical results.  If set to ``True``, a hint to the true value is returned,
+    even though the exact number is generally unknowable.  See each method's
```

Comment:
Instead of a "hint to the true number", can we say that these properties won't be knowable exactly until runtime in the presence of control flow, but that with `recurse=True`, we return a proxy value that attempts to weigh over the program's possible branches to represent a cost of a particular implementation of a circuit. Suggested re-wordings welcome :)

Reply:
Done in 61380dd9e.